### ISSUE TITLE
Suggestion: create an issue template for enhancements

### ISSUE THREAD

**Issue by LordFlashmeow** [SpeciesFileGroup/taxonworks#2361]
From what I can tell, the GH issue tracker is the place to document requests for enhancements/features. In these requests, the format of the title is usually "As a _user/dev/curator_, I want _blank_". 

It makes sense to me to have a separate issue template for these requests. This both encourages use of the title style and sets the label to "enhancement" instead of "bug" by default.

**Comment by mjy** [comment]
Let's mock the template here, perhaps it  should:

* [ ] have a default `As an X I need Y to do Z` format in the title
* [ ] encourage users to understand difference b/w this and the bug template?!
* [ ] have a link to best practices for telling a "recipe" style story ("when I do a, and I have b, then I see c, and ... etc." )
* [ ] include optional section for mockups/screenshots 

others?

**Comment by LordFlashmeow** [comment]
For feature requests, users should provide their use case so you can [avoid the X Y problem](https://meta.stackexchange.com/questions/66377/what-is-the-xy-problem)

**Comment by debpaul** [comment]
@LordFlashmeow writes: 

> For feature requests, users should provide their use case so you can [avoid the X Y problem](https://meta.stackexchange.com/questions/66377/what-is-the-xy-problem)

Lovely, thanks for this! Yes, always a critical communication waypoint: important to check with the person to get at exactly their understanding of the issue they are trying to solve or suggest and not jump to an immediate (and potentially erroneous) next step/s.

**Comment by debpaul** [comment]
Note @LordFlashmeow that GitHub permissions make it impossible to give everyone direct access to labels. But we can create additional templates, perhaps "not a bug" as one of them. So, enhancements and feature requests could go this route. (Other versioning software sometimes have more friendly issue-tagging than GitHub. (My favorite so far anyway: Bugzilla). Here's how GBIF presents options to submitters 
![image](https://user-images.githubusercontent.com/2990155/127012117-91a569c3-558b-49df-9440-b6c93060c575.png)

**Comment by LordFlashmeow** [comment]
Yeah, I thought it would be nice to not have all new issues marked as a bugs until someone with permissions fixes it.

**Comment by debpaul** [comment]
> 
> 
> Yeah, I thought it would be nice to not have all new issues marked as a bugs until someone with permissions fixes it.

We have done this in taxonworks_doc repository, where we added a template for contributing to the manual. 

Do you think "not a bug" would work? (as in everything else, to be further labeled by a developer?)

**Comment by LordFlashmeow** [comment]
I've seen some repos make a label like `to be triaged` to filter for new issues. This way different templates can still add relevant labels, but they also get marked for handling by a dev.

**Comment by debpaul** [comment]
@LordFlashmeow I'd like it if the submitter was empowered :-) at least a little bit. I like the layer that GBIF built on top (it feeds GitHub tickets). That way the tickets are at least somewhat sorted to start out with.

**Comment by LordFlashmeow** [comment]
Indeed. I'm pretty sure you can set the issue template to assign multiple labels at once (but I can't verify that)

**Comment by debpaul** [comment]
(Thinking outloud) I do know that you can link a Google Form to GitHub, and code a label into that so that if a user fills out the Google Form, the "new" label is applied. I wonder if this method could be extended (if using javascript) to allow a user to pick labels when creating the issue. (I don't know how Tim did it at GBIF).

**Comment by LordFlashmeow** [comment]
According to [the docs](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository), you can set multiple default labels for templates, so the enhancement template can get the `enhancement` label, and the bug template can get the `bug` label, and so on.

**Comment by mjy** [comment]
^ Yes, it's easy to setup a workflow in GH whereby any labels are applied to issues created from some template.

Side note, our Issue semantics are fairly well resolved: https://github.com/SpeciesFileGroup/taxonworks/labels.  In general we would like to use *less* rather than more labels as we move forward, as their use/meaning "drifts" as more are added.
